tests/test_input_manager.py

```diff
@@ -4779,3 +4799,245 @@ def test_extract_input_data_by_key_list_key_error(
+
+    mock_modifiable = mocker.patch(
+        "RUFAS.input_manager.InputManager._is_modifiable_during_runtime", return_value=modifiable
+    )
+
+    result = input_manager._check_modifiability(variable_name, metadata_properties, eager_termination, info_map)
+
+    mock_modifiable.assert_called_once_with(variable_name=variable_name, variable_properties=metadata_properties)
+
+    patch_om_error.assert_not_called()
+    patch_om_warning.assert_not_called()
+    assert result
+
+
+@pytest.mark.parametrize(
+    "variable_name,metadata_properties,eager_termination,info_map,modifiable",
+    [("test", {"test": 12}, True, {"class": "aa", "function": "bb"}, False)],
+)
+def test_check_modifiability_error(
+    variable_name: str,
+    metadata_properties: dict[Any, Any],
+    eager_termination: bool,
+    info_map: dict[Any, Any],
+    modifiable: bool,
+    mocker: MockerFixture,
+) -> None:
+    """Unit test for _check_modifiability to ensure right errors were thrown"""
+    input_manager = InputManager()
+    mock_add_error = mocker.patch("RUFAS.output_manager.OutputManager.add_error")
+
+    mock_modifiable = mocker.patch(
+        "RUFAS.input_manager.InputManager._is_modifiable_during_runtime", return_value=modifiable
+    )
+
+    try:
+        input_manager._check_modifiability(variable_name, metadata_properties, eager_termination, info_map)
+    except PermissionError as e:
+        mock_modifiable.assert_called_once_with(variable_name=variable_name, variable_properties=metadata_properties)
+
+        mock_add_error.assert_called_once()
+        mock_modifiable.assert_called_once_with(variable_name=variable_name, variable_properties=metadata_properties)
+        assert e.args[0] == f"IM Runtime Modification Error: {variable_name} is not modifiable during runtime."
+
+
+@pytest.mark.parametrize(
+    "variable_name,metadata_properties,eager_termination,info_map,modifiable",
+    [("test", {"test": 12}, False, {"class": "aa", "function": "bb"}, False)],
+)
+def test_check_modifiability_warning(
+    variable_name: str,
+    metadata_properties: dict[str, int],
+    eager_termination: bool,
+    info_map: dict[str, Any],
+    modifiable: bool,
+    mocker: MockerFixture,
+) -> None:
+    """Unit test for _check_modifiability to ensure right warnings were thrown"""
+    input_manager = InputManager()
+    mock_add_warning = mocker.patch("RUFAS.output_manager.OutputManager.add_warning")
+
+    mock_modifiable = mocker.patch(
+        "RUFAS.input_manager.InputManager._is_modifiable_during_runtime", return_value=modifiable
+    )
+
+    result = input_manager._check_modifiability(variable_name, metadata_properties, eager_termination, info_map)
+    mock_modifiable.assert_called_once_with(variable_name=variable_name, variable_properties=metadata_properties)
+    mock_add_warning.assert_called_once()
+    mock_modifiable.assert_called_once_with(variable_name=variable_name, variable_properties=metadata_properties)
```

Comment:
```suggestion
    mock_modifiable.assert_called_once_with(variable_name=variable_name, variable_properties=metadata_properties)
    mock_add_warning.assert_called_once()
```
Probably only needs to be called once?